sync/class.jetpack-sync-sender.php

```diff
@@ -44,11 +45,29 @@ protected function __construct() {
 	}
 
 	private function init() {
+		add_action( 'jetpack_sync_before_send_queue_sync', array( $this, 'maybe_set_user_from_token' ), 1 );
+		add_action( 'jetpack_sync_before_send_queue_sync', array( $this, 'maybe_clear_user_from_token' ), 20 );
 		foreach ( Jetpack_Sync_Modules::get_modules() as $module ) {
 			$module->init_before_send();
 		}
 	}
 
+	public function maybe_set_user_from_token() {
+		$jetpack = Jetpack::init();
+		if ( $jetpack->verify_xml_rpc_signature() ) {
```

Comment:
I think we should probably save the output of this call or only do it when we are under doing XMLRPC. 
Just to avoid possible false positives.

Reply:
RE "only do it when we are under doing XMLRPC" - It doesn't look like there is any kind of `is_xml_rpc()` method. `$jetpack->verify_xml_rpc_signature()` appears to check whether we are in xml rpc. RE "we should probably save the output of this call" - the method also appears to memoize/cache the results, see https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L5249, so the body of the method should execute at most once per request. For these reasons, it looks like this is already only happening when we are under XMLRPC, and the output is already effectively saved.

Reply:
From looking at the function if things work out the value gets cached. If they don't then we can go though a lot of checks before returning false. 

You can check against a constant `XMLRPC_REQUEST` to see if we are doing an xml-rpc request.